### DISCUSSION TITLE
How to automatically replace %20 in dynamic routing so when you go to link it uses dash like /sycamore-den

### DISCUSSION THREAD

**Discussion by half-metal** [sveltejs/kit#5738] [Q&A] [answered]
I ran across an issue where a name of a place would have a space in it, and when using dynamic routing `[places]/place` for something like "Sycamore Den", the url is going to be `http://site-name/place/Sycamore%20Den`. Which isn't great if you want to tell someone to go to the site and just add in name of place to the URL, and also messes up searching and indexing this URL. Is there some auto-magical method to replace this with a dash and have it work the same way in SvelteKit? So both `http://site-name/places/Sycamore-Den` and `http://site-name/places/Sycamore%20Den` would work the same?

Otherwise, I will store another column with a converted name for this. So the NAME column would have "Sycamore Den" and the URI_NAME column would have "Sycamore-Den"

Any other ideas that would be better so I don't need to store another column?

**Comment by aubergene** [comment, marked as answer]
I would create a slug and store and index that. The slugify function should be idempotent so you can then safely slugify the param before looking for it in the database.

https://www.npmjs.com/package/slugify

**Reply by half-metal** [reply]
Yes that works, store the original slug, then creating a function to convert with a replaceAll to pass on in page params, then reverting that param, then do database call for data. I then came to realize I am going to need an additional DB column though, since it's unlikely my "names" in database can be unique. So I have a "name" column that would have a record like "Sycamore Den" and then a unique "url_name" that would have "sycamoreden". This allows additional Sycamore Den records, which can then have "url_name" of "sycamoreden-2" or whatever want to use.
